### DISCUSSION TITLE
KID is not relied to locate a JWK in a JWKS and validate the JWT

### DISCUSSION THREAD

**Discussion by doyyan** [lestrrat-go/jwx#418] [General]
Hi,
It looks like the KID is not used to locate a JWK and the package simply iterates through all the JWKs in a JWKS and validates the JWT with each JWK.
https://github.com/lestrrat-go/jwx/blob/3bb9a18ceb5f45ffc4d0b837f85a5c3eb3f0bbfe/jws/jws.go#L343

Is this the expected behaviour?
Our system testers have raised the defect as the JWT being validated did not have a "kid" claim in its header that matched any of the KIDs of the JWKs.

**Comment by lestrrat** [comment]
Unfortunately, [that function has been removed in 1.1.0](https://github.com/lestrrat-go/jwx/blob/d01528f5e5ae3221a48f67570928867e98184bbd/Changes#L274), and it was removed for a reason.
I'd suggest you upgrade to a more recent version.

If you can't upgrade, please send me a PR. I don't intend to actively maintain older versions myself.
